api/server.js

```
@@ -1,11 +1,157 @@
 server.use(cors()); //allows app to use cross origin sharing
 server.use(express.json()); //tells Express to automatically parse incoming requests with Content-Type: application/json and put the parsed data on req.body
 
+const boardPrisma = require("./board-prisma-calls.js");
+const cardPrisma = require("./card-prisma-calls.js");
 
+//BELOW APIS ARE ALL CRUD FOR BOARDS
 
+server.get("/api/boards", async (req, res, next) => {
+  const search = req.query;
+  try {
+    // How we get the boards is inside the find function which uses prisma client
+    const boards = await boardPrisma.find(search);
+    if (boards.length) {
+      res.json(boards);
+    } else {
+      next({
+        status: 404,
+        message: "No boards found match the search criteria",
+      });
+    }
+  } catch (err) {
+    next(err);
+  }
+});
 
-module.exports = server
\ No newline at end of file
+server.post("/api/boards", async (req, res, next) => {
+  const newBoard = req.body; //save whatever is in req body to newBoard
+
+  try {
+    //validate that this newBoard object has fields required to succesfully make new board
+
+    const isNewBoardValid =
+      newBoard.title !== undefined && newBoard.category !== undefined;
+
+    if (isNewBoardValid) {
+      const createdBoard = await boardPrisma.create(newBoard);
+      res.status(201).json(createdBoard);
+    } else {
+      next({ status: 422, message: "board title and category are required" });
+    }
+  } catch (err) {
+    next(err);
+  }
+});
+
+server.delete("/api/boards/:id", async (req, res, next) => {
+  const id = Number(req.params.id); //get the id of the board that is being requested for deletion
+  try {
+    const boardToDelete = await boardPrisma.findById(id); //use find by id function to see if board requested to delete even exists
+    if (boardToDelete) {
+      const deletedBoard = await boardPrisma.delete(id);
+      res.json(deletedBoard);
+    } else {
+      next({ status: 404, message: "board not found" });
```

Sometimes if the board doesn't exist already, you can silently return instead of returning the 404, that will also depend on what do you want to do on the client side